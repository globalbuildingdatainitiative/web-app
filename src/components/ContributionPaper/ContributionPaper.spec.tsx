import { ContributionPaper } from '@components'
import { MockedProvider } from '@apollo/client/testing'
import { getContributionsMock } from '@mocks'
import { expect, render, test, screen } from '@test'

test('ContributionPaper', async () => {
  const { baseElement } = render(
    <MockedProvider mocks={getContributionsMock} addTypename={false}>
      <ContributionPaper />
    </MockedProvider>,
  )
  expect(await screen.findByTestId('ContributionPaper')).toBeTruthy()
  expect(baseElement).toMatchSnapshot()
})
