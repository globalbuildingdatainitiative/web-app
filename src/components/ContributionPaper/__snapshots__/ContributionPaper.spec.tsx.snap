// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContributionPaper 1`] = `
<body>
  <div>
    <style
      data-mantine-styles="true"
    >
      :root{--mantine-font-family: Outfit, sans-serif;--mantine-font-family-headings: Outfit, sans-serif;--mantine-breakpoint-xxl: 120em;}
    </style>
    <style
      data-mantine-styles="classes"
    >
      @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}@media (max-width: 119.99375em) {.mantine-visible-from-xxl {display: none !important;}}@media (min-width: 120em) {.mantine-hidden-from-xxl {display: none !important;}}
    </style>
    <div
      class="m_e615b15f mantine-Card-root m_1b7284a3 mantine-Paper-root"
      data-testid="ContributionPaper"
      data-with-border="true"
      style="--paper-radius: var(--mantine-radius-lg); --paper-shadow: var(--mantine-shadow-sm); --card-padding: var(--mantine-spacing-lg);"
    >
      <h3
        class="m_8a5d1357 mantine-Title-root"
        data-order="3"
        style="--title-fw: var(--mantine-h3-font-weight); --title-lh: var(--mantine-h3-line-height); --title-fz: var(--mantine-h3-font-size); margin-bottom: 8px;"
      >
        All Contributions
      </h3>
      <div
        data-testid="ContributionTable"
      >
        <div
          class="mrt-table-paper MRT_TablePaper-module_root__q0v5L m_1b7284a3 mantine-Paper-root"
          data-with-border="true"
          style="--paper-shadow: var(--mantine-shadow-xs);"
        >
          <div
            class="common-styles-module_common-toolbar-styles__DnjR8 MRT_TopToolbar-module_root__r4-V9"
            style=""
          >
            <div
              class="MRT_TopToolbar-module_actions-container__-uL0u m_8bffd616 mantine-Flex-root __m__-r4"
              style=""
            >
              <span />
              <div
                class="m_8bffd616 mantine-Flex-root __m__-r6"
                style="justify-content: end; flex-wrap: wrap-reverse;"
              >
                <div
                  aria-hidden="true"
                  class="MRT_GlobalFilterTextInput-module_collapse__v311d"
                  style="box-sizing: border-box; opacity: 0; transition: opacity 200ms ease; display: none; height: 0px; overflow: hidden;"
                >
                  <div
                    class="mrt-global-filter-text-input MRT_GlobalFilterTextInput-module_root__Xmcpv m_46b77525 mantine-InputWrapper-root mantine-TextInput-root"
                    data-variant="filled"
                    style="margin-top: 0rem;"
                  >
                    <div
                      class="m_6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
                      data-variant="filled"
                      data-with-left-section="true"
                      style=""
                    >
                      <div
                        class="m_82577fc2 mantine-Input-section mantine-TextInput-section"
                        data-position="left"
                        style=""
                      >
                        <svg
                          class="tabler-icon tabler-icon-search "
                          fill="none"
                          height="24"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M10 10m-7 0a7 7 0 1 0 14 0a7 7 0 1 0 -14 0"
                          />
                          <path
                            d="M21 21l-6 -6"
                          />
                        </svg>
                      </div>
                      <input
                        aria-invalid="false"
                        class="m_8fb7ebe7 mantine-Input-input mantine-TextInput-input"
                        data-variant="filled"
                        formaction="http://localhost:3000/"
                        formmethod=""
                        id=""
                        placeholder="Search"
                        style=""
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="mrt-toolbar-internal-buttons MRT_ToolbarInternalButtons-module_root__NKoUG m_8bffd616 mantine-Flex-root __m__-rd"
                  style=""
                >
                  <button
                    aria-label="Show/Hide search"
                    class="mantine-focus-auto mantine-active m_8d3f4000 mantine-ActionIcon-root m_87cf2631 mantine-UnstyledButton-root"
                    data-size="lg"
                    data-variant="subtle"
                    style="--ai-size: var(--ai-size-lg); --ai-bg: transparent; --ai-hover: var(--mantine-color-gray-light-hover); --ai-color: var(--mantine-color-gray-light-color); --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
                    type="button"
                  >
                    <span
                      class="m_8d3afb97 mantine-ActionIcon-icon"
                      style=""
                    >
                      <svg
                        class="tabler-icon tabler-icon-search "
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M10 10m-7 0a7 7 0 1 0 14 0a7 7 0 1 0 -14 0"
                        />
                        <path
                          d="M21 21l-6 -6"
                        />
                      </svg>
                    </span>
                  </button>
                  <button
                    aria-controls=""
                    aria-expanded="false"
                    aria-haspopup="menu"
                    aria-label="Show/Hide columns"
                    class="mantine-focus-auto mantine-active m_8d3f4000 mantine-ActionIcon-root m_87cf2631 mantine-UnstyledButton-root"
                    data-size="lg"
                    data-variant="subtle"
                    id=""
                    style="--ai-size: var(--ai-size-lg); --ai-bg: transparent; --ai-hover: var(--mantine-color-gray-light-hover); --ai-color: var(--mantine-color-gray-light-color); --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
                    type="button"
                  >
                    <span
                      class="m_8d3afb97 mantine-ActionIcon-icon"
                      style=""
                    >
                      <svg
                        class="tabler-icon tabler-icon-columns "
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M4 6l5.5 0"
                        />
                        <path
                          d="M4 10l5.5 0"
                        />
                        <path
                          d="M4 14l5.5 0"
                        />
                        <path
                          d="M4 18l5.5 0"
                        />
                        <path
                          d="M14.5 6l5.5 0"
                        />
                        <path
                          d="M14.5 10l5.5 0"
                        />
                        <path
                          d="M14.5 14l5.5 0"
                        />
                        <path
                          d="M14.5 18l5.5 0"
                        />
                      </svg>
                    </span>
                  </button>
                  <button
                    aria-label="Toggle density"
                    class="mantine-focus-auto mantine-active m_8d3f4000 mantine-ActionIcon-root m_87cf2631 mantine-UnstyledButton-root"
                    data-size="lg"
                    data-variant="subtle"
                    style="--ai-size: var(--ai-size-lg); --ai-bg: transparent; --ai-hover: var(--mantine-color-gray-light-hover); --ai-color: var(--mantine-color-gray-light-color); --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
                    type="button"
                  >
                    <span
                      class="m_8d3afb97 mantine-ActionIcon-icon"
                      style=""
                    >
                      <svg
                        class="tabler-icon tabler-icon-baseline-density-medium "
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M4 20h16"
                        />
                        <path
                          d="M4 12h16"
                        />
                        <path
                          d="M4 4h16"
                        />
                      </svg>
                    </span>
                  </button>
                  <button
                    aria-label="Toggle full screen"
                    class="mantine-focus-auto mantine-active m_8d3f4000 mantine-ActionIcon-root m_87cf2631 mantine-UnstyledButton-root"
                    data-size="lg"
                    data-variant="subtle"
                    style="--ai-size: var(--ai-size-lg); --ai-bg: transparent; --ai-hover: var(--mantine-color-gray-light-hover); --ai-color: var(--mantine-color-gray-light-color); --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
                    type="button"
                  >
                    <span
                      class="m_8d3afb97 mantine-ActionIcon-icon"
                      style=""
                    >
                      <svg
                        class="tabler-icon tabler-icon-maximize "
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M4 8v-2a2 2 0 0 1 2 -2h2"
                        />
                        <path
                          d="M4 16v2a2 2 0 0 0 2 2h2"
                        />
                        <path
                          d="M16 4h2a2 2 0 0 1 2 2v2"
                        />
                        <path
                          d="M16 20h2a2 2 0 0 0 2 -2v-2"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="MRT_ProgressBar-module_collapse__rOLJH MRT_ProgressBar-module_collapse-top__oCi0h"
              style="box-sizing: border-box; opacity: 0; transition: opacity 200ms ease; display: none; height: 0px; overflow: hidden;"
            >
              <div
                aria-busy="true"
                class="m_db6d6462 mantine-Progress-root"
                radius="0"
                style="--progress-radius: 0rem;"
              >
                <div
                  aria-label="Loading"
                  aria-valuemax="100"
                  aria-valuemin="0"
                  aria-valuenow="100"
                  aria-valuetext="100%"
                  class="m_2242eb65 mantine-Progress-section"
                  data-animated="true"
                  data-striped="true"
                  role="progressbar"
                  style="--progress-section-width: 100%; --progress-section-color: var(--mantine-color-blue-filled);"
                />
              </div>
            </div>
          </div>
          <div
            class="mrt-table-container MRT_TableContainer-module_root__JIsGB"
            style="--mrt-top-toolbar-height: 0;"
          >
            <table
              class="mrt-table MRT_Table-module_root__ms2uS m_b23fa0ef mantine-Table-table"
              style="--table-horizontal-spacing: var(--mantine-spacing-md); --table-vertical-spacing: var(--mantine-spacing-md); --header-id-size: 50; --col-id-size: 50; --header-uploadedAt-size: 50; --col-uploadedAt-size: 50; --header-organization_name-size: 300; --col-organization_name-size: 300; --header-project_name-size: 300; --col-project_name-size: 300;"
            >
              <thead
                class="m_b242d975 mantine-Table-thead MRT_TableHead-module_root__j9NkO MRT_TableHead-module_root-table-row-group__d9FO4"
                style="position: relative;"
              >
                <tr
                  class="m_4e7aa4fd mantine-Table-tr MRT_TableHeadRow-module_root__hUKv4"
                  data-hover="true"
                  data-with-row-border="true"
                  style=""
                >
                  <th
                    align="left"
                    class="m_4e7aa4f3 mantine-Table-th MRT_TableHeadCell-module_root__6y50a"
                    colspan="1"
                    data-index="0"
                    style="min-width: max(calc(var(--header-id-size) * 1px), 40px);"
                  >
                    <div
                      class="mrt-table-head-cell-content MRT_TableHeadCell-module_content__-pzSK m_8bffd616 mantine-Flex-root __m__-r1i"
                      style=""
                    >
                      <div
                        class="mrt-table-head-cell-labels MRT_TableHeadCell-module_labels__oiMSr MRT_TableHeadCell-module_labels-data__PvFGO m_8bffd616 mantine-Flex-root __m__-r1k"
                        style="--mrt-table-head-cell-labels-padding-left: 1.75;"
                      >
                        <div
                          class="mrt-table-head-cell-content-wrapper MRT_TableHeadCell-module_content-wrapper__py6aJ MRT_TableHeadCell-module_content-wrapper-hidden-overflow__QY40r MRT_TableHeadCell-module_content-wrapper-nowrap__-4aIg m_8bffd616 mantine-Flex-root __m__-r1m"
                          style=""
                        >
                          ID
                        </div>
                      </div>
                      <div
                        class="mrt-table-head-cell-content-actions MRT_TableHeadCell-module_content-actions__utxbm m_8bffd616 mantine-Flex-root __m__-r1o"
                        style=""
                      />
                    </div>
                  </th>
                  <th
                    align="left"
                    class="m_4e7aa4f3 mantine-Table-th MRT_TableHeadCell-module_root__6y50a"
                    colspan="1"
                    data-index="1"
                    style="min-width: max(calc(var(--header-uploadedAt-size) * 1px), 40px);"
                  >
                    <div
                      class="mrt-table-head-cell-content MRT_TableHeadCell-module_content__-pzSK m_8bffd616 mantine-Flex-root __m__-r1r"
                      style=""
                    >
                      <div
                        class="mrt-table-head-cell-labels MRT_TableHeadCell-module_labels__oiMSr MRT_TableHeadCell-module_labels-data__PvFGO m_8bffd616 mantine-Flex-root __m__-r1t"
                        style="--mrt-table-head-cell-labels-padding-left: 1.75;"
                      >
                        <div
                          class="mrt-table-head-cell-content-wrapper MRT_TableHeadCell-module_content-wrapper__py6aJ MRT_TableHeadCell-module_content-wrapper-hidden-overflow__QY40r MRT_TableHeadCell-module_content-wrapper-nowrap__-4aIg m_8bffd616 mantine-Flex-root __m__-r1v"
                          style=""
                        >
                          Date
                        </div>
                      </div>
                      <div
                        class="mrt-table-head-cell-content-actions MRT_TableHeadCell-module_content-actions__utxbm m_8bffd616 mantine-Flex-root __m__-r21"
                        style=""
                      />
                    </div>
                  </th>
                  <th
                    align="left"
                    class="m_4e7aa4f3 mantine-Table-th MRT_TableHeadCell-module_root__6y50a"
                    colspan="1"
                    data-index="2"
                    style="min-width: max(calc(var(--header-organization_name-size) * 1px), 40px);"
                  >
                    <div
                      class="mrt-table-head-cell-content MRT_TableHeadCell-module_content__-pzSK m_8bffd616 mantine-Flex-root __m__-r24"
                      style=""
                    >
                      <div
                        class="mrt-table-head-cell-labels MRT_TableHeadCell-module_labels__oiMSr MRT_TableHeadCell-module_labels-data__PvFGO m_8bffd616 mantine-Flex-root __m__-r26"
                        style="--mrt-table-head-cell-labels-padding-left: 1.75;"
                      >
                        <div
                          class="mrt-table-head-cell-content-wrapper MRT_TableHeadCell-module_content-wrapper__py6aJ MRT_TableHeadCell-module_content-wrapper-hidden-overflow__QY40r MRT_TableHeadCell-module_content-wrapper-nowrap__-4aIg m_8bffd616 mantine-Flex-root __m__-r28"
                          style=""
                        >
                          Organization
                        </div>
                      </div>
                      <div
                        class="mrt-table-head-cell-content-actions MRT_TableHeadCell-module_content-actions__utxbm m_8bffd616 mantine-Flex-root __m__-r2a"
                        style=""
                      />
                    </div>
                  </th>
                  <th
                    align="left"
                    class="m_4e7aa4f3 mantine-Table-th MRT_TableHeadCell-module_root__6y50a"
                    colspan="1"
                    data-index="3"
                    style="min-width: max(calc(var(--header-project_name-size) * 1px), 40px);"
                  >
                    <div
                      class="mrt-table-head-cell-content MRT_TableHeadCell-module_content__-pzSK m_8bffd616 mantine-Flex-root __m__-r2d"
                      style=""
                    >
                      <div
                        class="mrt-table-head-cell-labels MRT_TableHeadCell-module_labels__oiMSr MRT_TableHeadCell-module_labels-data__PvFGO m_8bffd616 mantine-Flex-root __m__-r2f"
                        style="--mrt-table-head-cell-labels-padding-left: 1.75;"
                      >
                        <div
                          class="mrt-table-head-cell-content-wrapper MRT_TableHeadCell-module_content-wrapper__py6aJ MRT_TableHeadCell-module_content-wrapper-hidden-overflow__QY40r MRT_TableHeadCell-module_content-wrapper-nowrap__-4aIg m_8bffd616 mantine-Flex-root __m__-r2h"
                          style=""
                        >
                          Project
                        </div>
                      </div>
                      <div
                        class="mrt-table-head-cell-content-actions MRT_TableHeadCell-module_content-actions__utxbm m_8bffd616 mantine-Flex-root __m__-r2j"
                        style=""
                      />
                    </div>
                  </th>
                </tr>
              </thead>
              <tbody
                class="m_b2404537 mantine-Table-tbody MRT_TableBody-module_root__kGhRy"
                style=""
              >
                <tr
                  class="m_4e7aa4fd mantine-Table-tr MRT_TableBodyRow-module_root__2c3D4"
                  data-hover="true"
                  data-index="0"
                  data-with-row-border="true"
                  style=""
                >
                  <td
                    class="m_4e7aa4ef mantine-Table-td MRT_TableBodyCell-module_root__Wf-zi MRT_TableBodyCell-module_root-data-col__HHcxc"
                    data-index="0"
                    data-last-row="true"
                    style="min-width: max(calc(var(--col-id-size) * 1px), 40px); --mrt-cell-align: left;"
                  >
                    e3b986a2-e135-42d7-a5fc-0063fde6e9b1
                  </td>
                  <td
                    class="m_4e7aa4ef mantine-Table-td MRT_TableBodyCell-module_root__Wf-zi MRT_TableBodyCell-module_root-data-col__HHcxc"
                    data-index="1"
                    data-last-row="true"
                    style="min-width: max(calc(var(--col-uploadedAt-size) * 1px), 40px); --mrt-cell-align: left;"
                  >
                    04/04/2018
                  </td>
                  <td
                    class="m_4e7aa4ef mantine-Table-td MRT_TableBodyCell-module_root__Wf-zi MRT_TableBodyCell-module_root-data-col__HHcxc"
                    data-index="2"
                    data-last-row="true"
                    style="min-width: max(calc(var(--col-organization_name-size) * 1px), 40px); --mrt-cell-align: left;"
                  />
                  <td
                    class="m_4e7aa4ef mantine-Table-td MRT_TableBodyCell-module_root__Wf-zi MRT_TableBodyCell-module_root-data-col__HHcxc"
                    data-index="3"
                    data-last-row="true"
                    style="min-width: max(calc(var(--col-project_name-size) * 1px), 40px); --mrt-cell-align: left;"
                  >
                    My Project
                  </td>
                </tr>
              </tbody>
              <tfoot
                class="mantine-Table-tfoot MRT_TableFooter-module_root__-JXpw"
                style=""
              />
            </table>
          </div>
          <div
            class="mrt-bottom-toolbar MRT_BottomToolbar-module_root__VDeWo common-styles-module_common-toolbar-styles__DnjR8"
            style=""
          >
            <div
              aria-hidden="true"
              class="MRT_ProgressBar-module_collapse__rOLJH"
              style="box-sizing: border-box; opacity: 0; transition: opacity 200ms ease; display: none; height: 0px; overflow: hidden;"
            >
              <div
                aria-busy="true"
                class="m_db6d6462 mantine-Progress-root"
                radius="0"
                style="--progress-radius: 0rem;"
              >
                <div
                  aria-label="Loading"
                  aria-valuemax="100"
                  aria-valuemin="0"
                  aria-valuenow="100"
                  aria-valuetext="100%"
                  class="m_2242eb65 mantine-Progress-section"
                  data-animated="true"
                  data-striped="true"
                  role="progressbar"
                  style="--progress-section-width: 100%; --progress-section-color: var(--mantine-color-blue-filled);"
                />
              </div>
            </div>
            <div
              class="MRT_BottomToolbar-module_custom-toolbar-container__XcDRF"
              style=""
            >
              <span />
              <div
                class="MRT_BottomToolbar-module_paginator-container__A3eWY"
                style=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    data-portal="true"
  />
  <div
    data-portal="true"
  />
  <div
    data-portal="true"
  />
  <div
    data-portal="true"
  />
  <div
    data-portal="true"
  />
</body>
`;
